scheduler-jobs/scheduler-jobs.go

```diff
@@ -0,0 +1,316 @@
+	for i := 0; i < 100; i++ {
+		name := "oneshot-job-" + strconv.Itoa(i)
+		req := &client.Job{
+			Name:     name,
+			Schedule: "@every 30s",
+			Repeats:  1, // one shot job
+			DueTime:  "5s",
+			TTL:      "40s",
+			Data:     nil,
+		}
+		err := daprClient.ScheduleJobAlpha1(context.Background(), req)
+		if err != nil {
+			log.Printf("Error scheduling oneshot job '%s': %s\n", name, err)
+		}
+	}
+}
+
+func scheduleIndefiniteJobs(daprClient client.Client) {
+	for i := 0; i < 100; i++ {
+		name := "indefinite-job-" + strconv.Itoa(i)
+		req := &client.Job{
+			Name:     name,
+			Schedule: "@every 30s",
+			DueTime:  "1s",
+			TTL:      "40s",
+			Data:     nil,
+		}
+		err := daprClient.ScheduleJobAlpha1(context.Background(), req)
+		if err != nil {
+			log.Printf("Error scheduling indefinite job '%s': %s\n", name, err)
+		}
+	}
+}
+
+func scheduleRepeatedJob(daprClient client.Client) {
+	name := "repeat-job"
+	req := &client.Job{
+		Name:     name,
+		Schedule: "@every 1s",
+		DueTime:  "0s",
+		Repeats:  maxRepeatTriggerCount,
+		TTL:      "10s",
+		Data:     nil,
+	}
+	err := daprClient.ScheduleJobAlpha1(context.Background(), req)
+	if err != nil {
+		log.Printf("Error scheduling repeat job '%s': %s\n", name, err)
+	}
+}
+
+func scheduleSingleJob(daprClient client.Client) {
+	name := "create-delete-job"
+	req := &client.Job{
+		Name:     name,
+		Schedule: "@every 1s",
+		DueTime:  "1s",
+		Repeats:  maxCreateDeleteTriggerCount,
+		TTL:      "3s",
+		Data:     nil,
+	}
+	err := daprClient.ScheduleJobAlpha1(context.Background(), req)
+	if err != nil {
+		log.Printf("Error scheduling single job '%s': %s\n", name, err)
+	}
+}
+
+func deleteSingleJob(daprClient client.Client) {
+	name := "create-delete-job"
+	err := daprClient.DeleteJobAlpha1(context.Background(), name)
+	if err != nil {
+		log.Printf("Error deleting single job '%s': %s\n", name, err)
+	}
+	createDelete.Store(0)
+	// signal to start the create/delete process again after waiting 2 minutes
+	createDeleteDoneCh <- struct{}{}
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	go func(ctx context.Context) {
+		if err := startServer(ctx); err != nil {
+			log.Fatalf("Error starting server: %v", err)
+		}
+	}(ctx)
+
+	// --dapr-grpc-port=3501
+	daprClient, err := client.NewClientWithPort(daprGRPCPort)
+	if err != nil {
+		log.Fatalf("Error getting dapr client: %v", err)
+	}
+	defer daprClient.Close()
+
+	// allow time for sidecar to connect otherwise if the job is scheduled beforehand then it gets dropped on the
+	// scheduler side bc there's no connection to send the job back on
+	log.Println("waiting a few seconds to let connections establish")
+	time.Sleep(5 * time.Second)
+
+	// Schedule initial batch of jobs
+	go scheduleOneshotJobs(daprClient)
+	go scheduleIndefiniteJobs(daprClient)
+	go scheduleRepeatedJob(daprClient)
+
+	// Schedule additional oneshot jobs once 100 are triggered
+	go func(ctx context.Context) {
+		select {
```

Comment:
Same for the other types.

Reply:
I don't follow your comment - that's what this test does. 

Initially, I schedule 100 jobs via `go scheduleOneshotJobs(daprClient)`. Then, there's a goroutine (the one you're highlighting) that indefinitely waits for a signal on the `oneshotDoneCh` channel. When it receives that signal — indicating all 100 jobs have triggered — it schedules another 100 jobs and resumes waiting. This process continues indefinitely, scheduling a new batch of 100 jobs every time the previous batch is completed.

Reply:
Yeah, the go routine will wait for the signal on the `oneshotDoneCh` channel, but when it gets it it will exit, it won't wait for the next signal, meaning we only schedule the 100 jobs twice. That's why I suggested an infinite for loop.